migration-engine/core/src/api.rs

```diff
@@ -161,4 +167,33 @@ impl<C: MigrationConnector> GenericApi for C {
             .instrument(tracing::info_span!("SchemaPush"))
             .await
     }
+
+    async fn migration_diff(&self, schema: &str) -> CoreResult<String> {
+        let (configuration, datamodel) = crate::parse_schema(schema)?;
+
+        let migration = self
+            .diff(
+                DiffTarget::Database,
+                DiffTarget::Datamodel((&configuration, &datamodel)),
+            )
+            .await?;
+
+        Ok(self.migration_summary(&migration))
+    }
```

Comment:
I think this should live in the `test-cli` crate, since all the methods are on the connector and it's exposed. Or `GenericApi` can expose the connector directly if it doesn't yet.

Reply:
Sir, yes sir!